app/services/SectionsProvider.scala

```diff
@@ -50,37 +52,29 @@ class SectionsProvider @Inject()(protected val cacheConnector: DataCacheConnecto
     }
 
   def sections(cache: CacheMap) : Seq[Section] = {
-      mandatorySections(cache) ++
-      dependentSections(cache)
+    mandatorySections(cache) ++ dependentSections(cache)
   }
 
   def sectionsFromBusinessActivities(activities: Set[BusinessActivity],
                                      msbServices: Option[BusinessMatchingMsbServices])
-                                    (implicit cache: CacheMap): Set[Section] =
+                                    (implicit cache: CacheMap): Seq[Section] = {
 
-    activities.foldLeft[Set[Section]](Set.empty) {
-      (m, n) => n match {
-        case AccountancyServices =>
-          m + Asp.section + Supervision.section
-        case ArtMarketParticipant =>
-          m + Amp.section
-        case EstateAgentBusinessService =>
-          m + EstateAgentBusiness.section
-        case HighValueDealing =>
-          m + Hvd.section
-        case MoneyServiceBusiness if msbServices.isDefined =>
-          m + Msb.section
-        case TrustAndCompanyServices =>
-          m + Tcsp.section + Supervision.section
-        case _ => m
-      }
+    val asp = if(activities.contains(AccountancyServices)) Seq(Asp.section) else Seq.empty
+    val tcsp = if(activities.contains(TrustAndCompanyServices)) Seq(Tcsp.section) else Seq.empty
+    val supervision = if(asp.nonEmpty || tcsp.nonEmpty) Seq(Supervision.section) else Seq.empty
+    val amp = if(activities.contains(ArtMarketParticipant)) Seq(Amp.section) else Seq.empty
+    val eab = if(activities.contains(EstateAgentBusinessService)) Seq(EstateAgentBusiness.section) else Seq.empty
+    val hvd = if(activities.contains(HighValueDealing)) Seq(Hvd.section) else Seq.empty
+    val msb = if(activities.contains(MoneyServiceBusiness) && msbServices.isDefined) Seq(Msb.section) else Seq.empty
+
+    asp ++ tcsp ++ supervision ++ amp ++ eab ++ hvd ++ msb
```

Comment:
just small explanation, why this rewrite. In old method with fold and matching order of services wasn't guaranteed.